File: frontend/src/types/readiness.ts
/** Readiness state shape returned by GET /readiness — v0.8.7 */
export interface ReadinessState {
  launcher_blocked: boolean
  setup_completed: boolean
  update_required: boolean
  preflight_passed: boolean
  backend_ready: boolean
  final_backend_ready: boolean
  release_ready: boolean
  live_applied_testing_ready: boolean
  blocked_reason_messages: string[]
  continue_destination: string | null
  frontend_port: number
  backend_port: number
  readiness_poll_interval_ms: number
}

File: backend/app/api/schemas/readiness.py
"""Readiness API schema — v0.8.7."""
from typing import List, Optional
from pydantic import BaseModel


class ReadinessResponse(BaseModel):
    launcher_blocked: bool
    setup_completed: bool
    update_required: bool
    preflight_passed: bool
    backend_ready: bool
    final_backend_ready: bool
    release_ready: bool
    live_applied_testing_ready: bool
    blocked_reason_messages: List[str]
    continue_destination: Optional[str]
    frontend_port: int
    backend_port: int
    readiness_poll_interval_ms: int
